plutus-tx-plugin/src/PlutusTx/Compiler/Type.hs

```diff
@@ -119,8 +120,7 @@ compileTyCon :: forall uni fun m ann. CompilingDefault uni fun m ann => GHC.TyCo
 compileTyCon tc
     | tc == GHC.intTyCon = throwPlain $ UnsupportedError "Int: use Integer instead"
     | tc == GHC.intPrimTyCon = throwPlain $ UnsupportedError "Int#: unboxed integers are not supported"
-    -- Surprisingly, `Void#` is actually more like `Unit` than `Void`, so we represent it as such.
-    | tc == GHC.voidPrimTyCon = pure (PIR.mkTyBuiltin @_ @() AnnOther)
+    | tc == GHC.unboxedUnitTyCon = pure (PIR.mkTyBuiltin @_ @() AnnOther)
```

Comment:
Reason for this change: starting GHC 9.2, `Void#` is a type synonym for `(##)`, rather than a separate type.

Reply:
This version makes more sense anyway!